### PR TITLE
recursively normalize keys

### PR DESCRIPTION
**Pull request by achille-roussel**
I still spotted a couple of invalid keys, it turns out nested properties also have to be normalized.

### CODE REVIEW

File: lib/mapper.js [hunk 1]
```diff
@@ -55,16 +55,43 @@ exports.track = function(track, settings) {
...
-    if (event.hasOwnProperty(name)) {
-      normalizedEvent[normalizeName(name)] = event[name];
+function normalize(value) {
+  if (value === undefined) {
+    return undefined;
+  }
+  if (value === null) {
+    return null;
+  }
+  if (value instanceof Array) {
```
Comment (stephenmathieson) [inline lib/mapper.js]: Doesn't look like we're adding a test case for this. Any chance you could add one?

Comment (achille-roussel) [inline lib/mapper.js]: We are, see `"Exception": [{ ...`

Comment (stephenmathieson) [inline lib/mapper.js]: Ah - nice, my bad.

**Comment by achille-roussel** [comment]
Thanks for the review!

**Pull request merged**